File: solution/2_Encodering/2_Encodering/LengteEncoderingAlgoritmeTest.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using encodering;

namespace test
{
    [TestClass]
    public class LengteEncoderingAlgoritmeTest
    {
        private LengteEncoderingAlgoritme lengteEncodering;

        [TestInitialize]
        public void setup() {
		    lengteEncodering = new LengteEncoderingAlgoritme();
	    }

        [TestMethod]
        public void legeString() {
		    Assert.AreEqual("", lengteEncodering.encodeer(""));
	    }

	    [TestMethod]
	    public void geenEncoderingNodig() {
		    Assert.AreEqual("abc", lengteEncodering.encodeer("abc"));
	    }

	    [TestMethod]
	    public void encodeerVolledigeString() {
		    Assert.AreEqual("3a", lengteEncodering.encodeer("aaa"));
	    }

	    [TestMethod]
	    public void encodeerEersteLetters() {
		    Assert.AreEqual("3abc", lengteEncodering.encodeer("aaabc"));
	    }

	    [TestMethod]
	    public void encodeerLettersInMidden() {
		    Assert.AreEqual("a4bc", lengteEncodering.encodeer("abbbbc"));
	    }

	    [TestMethod]
	    public void encodeerLettersOpEinde() {
		    Assert.AreEqual("ab2c", lengteEncodering.encodeer("abcc"));
	    }

	    [TestMethod]
	    public void meerdereAaneenschakelingenVanHetzelfdeKarakter() {
		    Assert.AreEqual("4ab2ac5a", lengteEncodering.encodeer("aaaabaacaaaaa"));
	    }

        [TestMethod]
	    public void meerdereAaneenschakelingenVanVerschillendeKarakters() {
		    Assert.AreEqual("4ab3c", lengteEncodering.encodeer("aaaabccc"));
	    }
    }
}

File: workspace/2_Encodering/test/encodering/LengteEncoderingAlgoritmeTest.java
package encodering;

import static junit.framework.Assert.assertEquals;

import org.junit.Before;
import org.junit.Test;

public class LengteEncoderingAlgoritmeTest {

	private LengteEncoderingAlgoritme lengteEncodering;

	@Before
	public void setup() throws Exception {
		lengteEncodering = new LengteEncoderingAlgoritme();
	}

	@Test
	public void legeString() {
		assertEquals("", lengteEncodering.encodeer(""));
	}

	@Test
	public void geenEncoderingNodig() {
		assertEquals("abc", lengteEncodering.encodeer("abc"));
	}

	@Test
	public void encodeerVolledigeString() {
		assertEquals("3a", lengteEncodering.encodeer("aaa"));
	}

	@Test
	public void encodeerEersteLetters() {
		assertEquals("3abc", lengteEncodering.encodeer("aaabc"));
	}

	@Test
	public void encodeerLettersInMidden() {
		assertEquals("a4bc", lengteEncodering.encodeer("abbbbc"));
	}

	@Test
	public void encodeerLettersOpEinde() {
		assertEquals("ab2c", lengteEncodering.encodeer("abcc"));
	}

	@Test
	public void meerdereAaneenschakelingenVanHetzelfdeKarakter() {
		assertEquals("4ab2ac5a", lengteEncodering.encodeer("aaaabaacaaaaa"));
	}

	@Test
	public void meerdereAaneenschakelingenVanVerschillendeKarakters() {
		assertEquals("4ab3c", lengteEncodering.encodeer("aaaabccc"));
	}
}
